### DISCUSSION TITLE
google/protobuf/runtime_version.h file not found

### DISCUSSION THREAD

**Discussion by ValeV008** [foxglove/community#1025] [General]
Hi,

I have an issue with changing example_server_protobuf.cpp to transmit .proto messages that I've defined.

I used protoc to create, and then copied _myProto.pb.h_ and _myProto.pb.cc_ files to _foxglove-websocket/include/foxglove_ folder. Then I added `#include "foxglove/myProto.pb.h"` to _example_server_protobuf.cpp_. When I run `nmake build`, I get error _'google/protobuf/runtime_version.h' file not found_ in _myProto.pb.h_.

How can I solve this error and what is the correct way to change the example_server_protobuf.cpp?

Thanks!

**Comment by defunctzombie** [comment]
Your build is not configured in a way to find the `protobuf` include file. Without knowing how you've structured the build setup it is hard to say more. Are you using the provided CMakeLists file in the examples repo?

**Comment by ValeV008** [comment]
Thanks for the response. Yes I use the CMakeLists from the examples. I noticed that the included protobuf library (from foxglove-websocket::foxglove-websocket) uses version 3.21.12, so I downloaded that version of protoc compiler, compiled my .proto file again and included new .pb.cc and pb.h files in the project. Now the project compiles fine. So I'm guessing it was version mismatch.

Btw, why does foxglove-websocket use such old protobuf lib version?

**Reply by defunctzombie** [reply]
_why_ is probably because we've not bothered to update the example to a newer version. If you want to make a PR updating the version and making sure the example works we'd be happy to review.

**Reply by ValeV008** [reply]
I understand. I will give it a try to update the examples. Thanks!
